File: client/android/div/src/main/java/com/yandex/div/core/util/mask/FixedLengthInputMask.kt
package com.yandex.div.core.util.mask

internal open class FixedLengthInputMask(
    initialMaskData: MaskData,
    private val onError: ((Exception) -> Unit)
) : BaseInputMask(initialMaskData) {
    override fun onException(exception: Exception) {
        onError.invoke(exception)
    }
}

File: client/web/divkit/src/utils/mask/fixedLengthInputMask.ts
import type { LogError, WrappedError } from '../wrapError';
import { BaseInputMask, MaskData } from './baseInputMask';

export class FixedLengthInputMask extends BaseInputMask {
    constructor(initialMaskData: MaskData, private readonly logError: LogError) {
        super(initialMaskData);
    }

    onException(exception: WrappedError) {
        this.logError(exception);
    }
}
